core/modules/loader/CentralClient.cc

```diff
@@ -233,30 +233,28 @@ void CentralClient::_keyInsertReq(CompositeKey const& key, int chunk, int subchu
     } catch (boost::system::system_error e) {
         LOGS(_log, LOG_LVL_ERROR, "CentralClient::_keyInsertReq boost system_error=" << e.what() <<
                                   " key=" << key << " chunk=" << chunk << " sub=" << subchunk);
-        exit(-1); // TODO:&&& The correct course of action is unclear and requires thought,
-                  //       so just blow up so it's unmistakable something bad happened for now.
     }
 }
 
 
-KeyInfoData::Ptr CentralClient::keyInfoReq(CompositeKey const& key) {
+KeyInfoData::Ptr CentralClient::keyLookReq(CompositeKey const& key) {
     // Returns a pointer to a Tracker object that can be used to track job
     // completion and job status. keyInsertOneShot will call _keyInsertReq until
     // it knows the task was completed. _handleKeyInfoComplete marks
     // the jobs complete as the messages come in from workers.
     // Insert a oneShot DoListItem to keep trying to add the key until
     // we get word that it has been added successfully.
     LOGS(_log, LOG_LVL_INFO, "Trying to lookup key=" << key);
-    auto keyInfoOneShot = std::make_shared<CentralClient::KeyInfoReqOneShot>(this, key);
+    auto keyLookOneShot = std::make_shared<CentralClient::KeyLookReqOneShot>(this, key);
     {
-        std::unique_lock<std::mutex> lck(_waitingKeyInfoMtx);
+        std::unique_lock<std::mutex> lck(_waitingKeyLookMtx);
         // Limit the number of concurrent lookups.
         // If the key is already in the map, there is no point in blocking.
         int loopCount = 0;
-        auto iter = _waitingKeyInfoMap.find(key);
-        while (_waitingKeyInfoMap.size() > _doListMaxLookups
-               && iter == _waitingKeyInfoMap.end()) {
-            size_t sz = _waitingKeyInfoMap.size();
+        auto iter = _waitingKeyLookMap.find(key);
+        while (_waitingKeyLookMap.size() > _doListMaxLookups
+               && iter == _waitingKeyLookMap.end()) {
+            size_t sz = _waitingKeyLookMap.size();
             lck.unlock();
             if (loopCount % 100 == 0) {
```

Comment:
This condition may result in non-deterministic behavior as it depends on a value of `_maxRequestSleepTime` which can be changed w/o taking into account this loop . I would rather count the amount of time waited in `usleep(_maxRequestSleepTime)` and print the message, say every N seconds. Then you'd be getting periodic printouts every N seconds regardless of what you choose to put into `_maxRequestSleepTime`.